seed/views/v3/analyses.py

```diff
@@ -300,176 +298,32 @@ def stats(self, request):
-                elif record["gross_floor_area"] <= 25000:
-                    updated_record["gross_floor_area"] = "10,000-25,000"
-                elif record["gross_floor_area"] <= 50000:
-                    updated_record["gross_floor_area"] = "25,000-50,000"
-                elif record["gross_floor_area"] <= 100000:
-                    updated_record["gross_floor_area"] = "50,000-100,000"
-                elif record["gross_floor_area"] <= 200000:
-                    updated_record["gross_floor_area"] = "100,000-200,000"
-                elif record["gross_floor_area"] <= 500000:
-                    updated_record["gross_floor_area"] = "200,000-500,000"
-                elif record["gross_floor_area"] <= 1000000:
-                    updated_record["gross_floor_area"] = "500,000-1,000,000"
-                else:
-                    updated_record["gross_floor_area"] = "> 1,000,000"
-            square_footage_agg.append(updated_record)
-
-        g = defaultdict(int)
-        for h in square_footage_agg:
-            g[h["gross_floor_area"]] += h["count"]
-        square_footage_list2 = [
-            {"gross_floor_area": gross_floor_area, "percentage": count / views.count() * 100} for gross_floor_area, count in g.items()
-        ]
-
-        extra_data_list = []
-        for data in states.values_list("extra_data", flat=True):
-            for key, value in data.items():
-                extra_data_list.append(key)
-
-        count_agg = []
-        for i in col_names:
-            count_dict = {}
-            if i in column_names_extra_data:
-                count_dict[i] = len(list(filter(None, states.values_list("extra_data__" + i, flat=True))))
+        num_of_nonnulls_by_column_name = {}
+        for column in columns:
+            name = column.column_name
+            if column.is_extra_data:
+                count = states.filter(extra_data__has_key=name).exclude(**{f"extra_data__{name}": None}).count()
             else:
-                count_dict[i] = len(list(filter(None, states.values_list(i, flat=True))))
-            count_agg.append(count_dict)
+                count = states.exclude(**{name: None}).count()
 
-        count_agg_dict = {}
-        for d in count_agg:
-            count_agg_dict.update(d)
+            num_of_nonnulls_by_column_name[name] = count
 
         return JsonResponse(
             {
                 "status": "success",
                 "total_records": views.count(),
-                "number_extra_data_fields": len(extra_data_count),
-                "column_settings fields and counts": count_agg_dict,
-                "existing_extra_data fields and count": extra_data,
-                "property_types": property_types,
-                "year_built": year_built_list,
-                "energy": energy_list2,
-                "square_footage": square_footage_list2,
+                "number_extra_data_fields": columns.filter(is_extra_data=True).count(),
```

Comment:
this field has changed a little.
it used to be number of extra data columns with at least one appearance in a state in this cycle, now it's just number of extra data columns, period.